wled00/FX_fcn.cpp

```diff
@@ -174,12 +177,15 @@ void Segment::deallocateData() {
   * may free that data buffer.
   */
 void Segment::resetIfRequired() {
-  if (reset) {
-    deallocateData();
-    next_time = 0; step = 0; call = 0; aux0 = 0; aux1 = 0;
-    if (_queuedChanges) setUp(_qStart, _qStop, grouping, spacing, offset, _qStartY, _qStopY, true);
-    reset = false; // setOption(SEG_OPTION_RESET, false);
+  if (!reset) return;
+  
+  deallocateData();
+  next_time = 0; step = 0; call = 0; aux0 = 0; aux1 = 0;
+  if (_queuedChangesSegId == strip.getCurrSegmentId()) { // apply queued changes
```

Comment:
I know this looks ok, but it is not.
It mixes global `strip` into independent Segment class. IMO this is unacceptable (separation of functionality). I put a lot of effort to make Segment independent of WS2812FX and not mix code as it becomes unmanageable otherwise in the long term..
If you need to maintain ID to determine which segment is being worked on then add ID variable to Segment class.

For a better solution see another comment below.

Reply:
Hi @Aircoookie, unfortunately with your last commit "Static queued segment bounds" my crashes are coming back.

Not sure which code lines are to blame, as the crash dump does not give me any meaningful information. Usually it means we have memory corruption somewhere...

Reply:
That's unfortunate... please retest with the latest commit, although I wouldn't expect it to fix the crashes. What do I have to do to reproduce the issue? Just switching between presets with different multi-segment setups?

Reply:
Yes, to create a crash basicially I'm switching between several presets that have multiple segments. The presets segment layout should differ, this help to "enforce" the problem.

I can test again later this evening - currently busy@job.